app/src/main/java/app/grapheneos/camera/ui/composable/screen/GalleryScreen.kt

```diff
@@ -0,0 +1,349 @@
+                Toast.makeText(context, R.string.empty_gallery, Toast.LENGTH_LONG).show()
+                onExit()
+            }
+
+            capturedItems.clear()
+            capturedItems.addAll(items)
+        }
+    }
+
+
+    // Displays media info dialog when displayedMediaInfo is not null
+    MediaInfoDialog(
+        mediaItemDetails = displayedMediaInfo.value,
+        onFinish = {
+            displayedMediaInfo.value = null
+        },
+    )
+
+    // Displays item deletion dialog when deletionItem is not null
+    FileDeletionDialog(
+        deletionItem = itemToBeDeleted,
+        onDeleteAction = { item ->
+            coroutineScope.launch {
+                val result = item.delete(context)
+
+                if (result) {
+                    capturedItems.remove(item)
+
+                    if (capturedItems.isEmpty()) {
+                        Toast.makeText(context, R.string.empty_gallery, Toast.LENGTH_LONG)
+                            .show()
+                        onExit()
+                    } else {
+                        snackbarHostState.showOrReplaceSnackbar(
+                            context.getString(R.string.deleted_successfully)
+                        )
+                    }
+                } else {
+
+                    snackbarHostState.showOrReplaceSnackbar(
+                        context.getString(R.string.deleting_unexpected_error)
+                    )
+                }
+            }
+        },
+        dismissHandler = {
+            itemToBeDeleted = null
+        }
+    )
+
+    MaterialTheme(
+        colorScheme = MaterialTheme.colorScheme.copy(surface = Color.DarkGray)
+    ) {
+        Scaffold(
+            containerColor = backgroundColor,
+            contentWindowInsets = WindowInsets(0, 0, 0, 0),
+            snackbarHost = {
+                SnackbarHost(hostState = snackbarHostState)
+            },
+
+            topBar = {
+                GalleryTopBar(
+                    visible = !inFocusMode,
+                    onCloseAction = {
+                        onExit()
+                    },
+                    onEditAction = { chooseApp ->
+                        coroutineScope.launch {
+                            if (isSecureMode) {
+                                snackbarHostState.showOrReplaceSnackbar(
+                                    context.getString(R.string.edit_not_allowed)
+                                )
+                                return@launch
+                            }
+
+                            val currentItem = getCurrentMediaItem()
+
+                            val editIntent = Intent(Intent.ACTION_EDIT).apply {
+                                setDataAndType(currentItem.uri, currentItem.mimeType())
+                                putExtra(Intent.EXTRA_STREAM, currentItem.uri)
+                                addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION or Intent.FLAG_GRANT_WRITE_URI_PERMISSION)
```

Comment:
Why is `Intent.FLAG_GRANT_WRITE_URI_PERMISSION` added here? It wasn't used by the previous implementation for this intent.

Reply:
It was initially added to test whether the image updates after editing it from another app (and fixes around it). This issue still exists due to the limited options present in the library (to re-compose the image) after returning back to the app. I have added a separate toggle for in-place editing in the local commits. Will update this thread as soon as the main issue gets fixed and the changes are pushed

Reply:
Have updated the code to ensure things work as expected. A separate option is now present to support editing the original image while the behavior of the original options has been set as before